### PR TITLE
Latest changes from ROCM doc

### PR DESCRIPTION
**Pull request by anubhavamd**
@amathews-amd can you please look at this

### CODE REVIEW

File: Deep_learning/Deep-learning.rst [hunk 1]
```diff
@@ -290,22 +294,23 @@ Recommended: Install using published PyTorch ROCm docker image:
...
-   
+   docker pull rocm/pytorch:latest   
 
 
 3. Start a docker container using the downloaded image:
 
 ::
-
+  docker run -it --privileged --device=/dev/kfd --device=/dev/dri --group-add video --ipc=host --shm-size 8G rocm/pytorch:latest
   sudo docker run -it -v $HOME:/data --privileged --rm --device=/dev/kfd --device=/dev/dri --group-add video rocm/pytorch:rocm3.7_ubuntu16.04_py3.6_pytorch
```
Comment (amathews-amd) [inline Deep_learning/Deep-learning.rst:304]: Missed deleting original line.

File: Deep_learning/Deep-learning.rst [hunk 2]
```diff
@@ -290,22 +294,23 @@ Recommended: Install using published PyTorch ROCm docker image:
...
 
 
 4. Confirm working installation:
 
 ::
 
-  PYTORCH_TEST_WITH_ROCM=1 python3.6 test/run_test.py --verbose
+  PYTORCH_TEST_WITH_ROCM=1 
+  python3.6 pytorch/test/run_test.py
```
Comment (amathews-amd) [inline Deep_learning/Deep-learning.rst:312]: This addition seems wrong. the original changes seems right.

File: Deep_learning/Deep-learning.rst [hunk 3]
```diff
@@ -274,6 +274,10 @@ This is a quick guide to setup PyTorch with ROCm support inside a docker contain
 
 Note: Currently, ROCm install version 3.3 is required.
 
+
+*********
+Option 1
```
Comment (amathews-amd) [inline Deep_learning/Deep-learning.rst:279]: Formating of 'option 1', 'option 2', 'option 3' must be same ? Add title for option 1.

File: Deep_learning/Deep-learning.rst [hunk 4]
```diff
@@ -318,80 +323,77 @@ Recommended: Install using published PyTorch ROCm docker image:
 This step is optional but most PyTorch scripts will use torchvision to load models. E.g., running the pytorch examples requires torchvision.
 
 
-Option 2: Install using PyTorch upstream docker file
+Option 2: Install PyTorch using PyTorch ROCm base docker image
```
Comment (amathews-amd) [inline Deep_learning/Deep-learning.rst:326]: change title to 'Install using base docker image'

**Comment by amathews-amd** [comment]
cc: @jithunnair-amd @jeffdaily

File: Deep_learning/Deep-learning.rst [hunk 5]
```diff
@@ -318,80 +323,77 @@ Recommended: Install using published PyTorch ROCm docker image:
...
-::
+4.	Build PyTorch for ROCm:
+By default, PyTorch will build for gfx900, gfx906 and gfx908 simultaneously. To see which AMD uarch you have, run rocminfo | grep gfx (might need to install rocminfo package). 
+If you want to compile only for your uarch, 
+export PYTORCH_ROCM_ARCH=<uarch> 
+where <uarch> is the architecture reported by the rocminfo command. 
+Then build with
+.jenkins/pytorch/build.sh
 
-  PYTORCH_TEST_WITH_ROCM=1 python test/run_test.py --verbose
```
Comment (amathews-amd) [inline Deep_learning/Deep-learning.rst:375]: Where is testing for option 2 and option 3 ?

File: Deep_learning/Deep-learning.rst [hunk 6]
```diff
@@ -318,80 +323,77 @@ Recommended: Install using published PyTorch ROCm docker image:
...
+.jenkins/pytorch/test.sh
+This runs all CI unit tests and skips as appropriate on your system based on ROCm and, e.g., single or multi GPU configuration. No tests will fail if the compilation and installation is correct. Additionally, this step will install/update a supported version of torchvision, which most PyTorch scripts use to load models. 
+Individual test sets can be run with:
+PYTORCH_TEST_WITH_ROCM=1 python3 test/test_nn.py --verbose
+where test_nn.py can be replaced with any other test set.
 
-Option 3: Install using minimal ROCm docker file
-************************************************
 
 1. Download dockerfile based on the OS choose:
```
Comment (amathews-amd) [inline Deep_learning/Deep-learning.rst:398]: Why is the old content in option 3 (below) kept ?

File: Deep_learning/Deep-learning.rst [hunk 7]
```diff
@@ -270,230 +270,196 @@ PyTorch
...
+Using Docker gives you portability and access to a pre-built docker container that has been rigorously tested within AMD. This might also save on the compilation time and should perform exactly as it did when tested, without facing potential installation issues.
 
-::
+1. Pull the latest public PyTorch Docker image 
 
-  sudo apt-get install rocm-dev
-  or
-  sudo apt-get update
-  sudo apt-get upgrade
+docker pull rocm/pytorch:latest
```
Comment (amathews-amd) [inline Deep_learning/Deep-learning.rst:281]: script mode using backticks (left of 1 key on keyboard)
`docker pull rocm/pytorch:latest`

File: Deep_learning/Deep-learning.rst [hunk 8]
```diff
@@ -270,230 +270,196 @@ PyTorch
...
 
-3. Start a docker container using the downloaded image:
+Option 2: Install PyTorch Using Wheels Package
+**********************************************
 
-::
+PyTorch supports the ROCm platform by providing tested wheel packages. To access this feature, refer to https://pytorch.org/get-started/locally/, and choose the ‘ROCm’ compute platform. 
 
-  sudo docker run -it -v $HOME:/data --privileged --rm --device=/dev/kfd --device=/dev/dri --group-add video rocm/pytorch:rocm3.7_ubuntu16.04_py3.6_pytorch
+1. Obtain a base docker image with the correct user-space ROCm version installed from https://hub.docker.com/repository/docker/rocm/rocm-terminal or download a base OS docker image and install ROCm following the installation directions in Section 2.2.3. In this example, ROCm 4.2 is installed, as supported by the installation matrix from the pytorch.org website.
```
Comment (amathews-amd) [inline Deep_learning/Deep-learning.rst:298]: Remove section 2.2.3, reference in markdown

File: Deep_learning/Deep-learning.rst [hunk 9]
```diff
@@ -270,230 +270,196 @@ PyTorch
...
 
-5. Install torchvision:
+4. Install torch and torchvision as specified by the installation matrix.
 
-::
+pip3 install torch torchvision==0.11.1 -f 
+https://download.pytorch.org/whl/rocm4.2/torch_stable.html
 
-  pip install torchvision
+Option 3: Install PyTorch Using PyTorch ROCm Base Docker Image
```
Comment (amathews-amd) [inline Deep_learning/Deep-learning.rst:317]: Option3, Option4 header formatting.

File: Deep_learning/Deep-learning.rst [hunk 10]
```diff
@@ -1095,29 +1061,3 @@ Release Notes
     * Protobuf - `V3.5.2 <https://github.com/protocolbuffers/protobuf/releases/tag/v3.5.2>`_
```
Comment (amathews-amd) [inline Deep_learning/Deep-learning.rst:1061]: Check with chris austen about MIGraphX, MIVisionX

**Comment by Rmalavally** [comment]
Thanks, Aswin.

File: Deep_learning/Deep-learning.rst (hunk 10 above)
Comment (causten) [reply Deep_learning/Deep-learning.rst:1061]: there is nothing about migraphx in this file.  MIVisionX is owned by... 
kiriti.nageshgowda@amd.com

Rajy.MeeyakhanRawther@amd.com

**Pull request merged**